### ISSUE TITLE
Check correct origin not working with cdn url

### ISSUE THREAD

**Issue by dennis-eriksson-nethouse** [futurice/metalsmith-prismic-server#22]
req.body.apiUrl !== config.prismicUrl will not work when the cdn url changes from "https://your-project.prismic.io/api" to "https://your-project.cdn.prismic.io/api". The problem should be found in the file "build-server.js".

Am I doing it wrong or is there a way around this?

**Comment by doublejosh** [comment]
Prismic just announced a change forcing all users to use their CDN for API URLs. I assume this message is just forewarning you about the impending deprecation.

I've had this switched on all along and I haven't seen any such message. You will probably just need to update your `PRISMIC_URL` variable found in the `.env` file (if you used [metalsmith-prismic-template](https://github.com/futurice/metalsmith-prismic-template)).

I recommend you go switch on the CDN now and make the change before it happens underneath you on March 12.

_Here's a screenshot of the email they sent..._

<image width="500" src="https://d2ffutrenqvap3.cloudfront.net/items/0J1R1L1H2E2A0n1X0r2w/Screenshot%202018-03-01%2018.54.22.png?v=e2361066">

**Comment by dennis-eriksson-nethouse** [comment]
We have enabled the CDN for our repo on Prismic. I think that the problem is that when the build method in build-server.js tries to check correct origin(req.body.apiUrl), which I guess is the url from Prismic "https://your-project.prismic.io"(the client triggering the webhook). That url is not matching the new cdn url "https://your-project.cdn.prismic.io", this is why i think this req.body.apiUrl !== config.prismicUrl returns this **error** message: _Build rejected: mismatching api url_. 

Even though it seems like Prismic will redirect the urls after the 12th of March i would like to go directly to the new cdn url to be on the safe side.

**Comment by doublejosh** [comment]
Good call. Maybe report back if the message goes away after updating the `PRISMIC_URL`.

**Comment by dennis-eriksson-nethouse** [comment]
Hi, sorry, i was unclear. We have set the PRISMIC_URL to the new CDN route. That is why I think the whole problem arises. 

In the build method (/build) there is a validation that checks that the client request(Prismic triggering the webhook) has the same **host** url as the PRISMIC_URL which it does not, the url from Prismic is https://your-project.prismic.io and the url set to the PRISMIC_URL is https://your-project.cdn.prismic.io. This validation checks that the webhook only can be triggered from the owner of the Prismic Repo and not from any other domain. But now when i change to the CDN url the two urls does not match any more and the method jumps out and returns the error message that i earlier described. 

Note to say, we have only tried to manually trigger the webhook under the "settings" tab in Prismic(which has the url in the browser without the cdn).
